### ISSUE TITLE
Terra Date Picker isn't focusable or actionable with a screen reader.

### ISSUE THREAD

**Issue by JakeLaCombe** [cerner/terra-framework#464]
# Bug Report

## Description
Terra Datepicker is hard to use with a screen reader.

1. Focusing and clicking the button doesn't auto focus to the datepicker.
2. The calendar doesn't have descriptive text when used with a screen reader.

# Additional Details
It is discussed that the accessible workflow would be just having them use the text input. Perhaps this could be a solution?

**Comment by bjankord** [comment]
I've split this up into smaller issues as we get ready to start with this work. 
* https://github.com/cerner/terra-framework/issues/630
* https://github.com/cerner/terra-framework/issues/631
* https://github.com/cerner/terra-framework/issues/632
* https://github.com/cerner/terra-framework/issues/633
